openmdao/core/group.py

```diff
@@ -887,9 +887,15 @@ def _setup_global_connections(self, recurse=True, conns=None):
 
             if (prom_in not in allprocs_prom2abs_list_in and
                     prom_in not in self._var_allprocs_discrete['input']):
-                raise NameError(
-                    "%s: Input '%s' does not exist for connection from '%s' to '%s'." %
-                    (self.msginfo, prom_in, prom_out, prom_in))
+                if (prom_in in allprocs_prom2abs_list_out or
+                        prom_in in self._var_allprocs_discrete['output']):
+                    raise NameError(
+                        "%s: Attempted to connect to output '%s' from output '%s'." %
```

Comment:
Maybe add another sentence, something like "the target in 'connect' must be an input."

Reply:
fixed